cmd/aws/secrets_manager.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2025 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+)
+
+// TODO: Move ECDSAWithSHA256Signer to internal signer package.
```

Comment:
90% of the code in this file is the same as https://github.com/transparency-dev/static-ct/blob/2cebfbf1058abe79c52840a10c0fd9af43594a0a/cmd/gcp/secret_manager.go#L34.

Could we have an internal library with the shared code, and then code directly inside the main.go files the bits that fetch keys from the respective Secret Managers? We could leave them in a secret_manager.go file, but since it's less than 20 lines of code for each implementation, it might not be worth it.

Reply:
Sure. That's what I planned to do. It's not too urgent, we can leave it as-is and revisit this later.

Reply:
nit: AWS calls it "Secrets Manager" while GCP calls it "Secret Manager". How should we name this internally?